src/main/java/net/neoforged/neoforge/capabilities/CapabilityHooks.java

```diff
@@ -132,6 +135,11 @@ public static void registerVanillaProviders(RegisterCapabilitiesEvent event) {
                 Items.RED_SHULKER_BOX,
                 Items.WHITE_SHULKER_BOX,
                 Items.YELLOW_SHULKER_BOX);
+
+        // Colorable
+        event.registerEntity(Capabilities.Colorable.ENTITY_DYE, EntityType.SHEEP, (sheep, ctx) -> new SheepWoolDyeColorable(sheep));
+        event.registerEntity(Capabilities.Colorable.ENTITY_DYE, EntityType.CAT, (cat, ctx) -> new CatCollarDyeColorable(cat));
```

Comment:
Probably best to only return a capability for dogs and cats if tamed, otherwise null